packages/flutter_tools/lib/src/windows/build_windows.dart

```diff
@@ -98,6 +103,17 @@ Future<void> buildWindows(WindowsProject windowsProject, BuildInfo buildInfo, {
     status.stop();
   }
 
+  final Directory oldRunnerDirectory = buildDirectory
+    .parent
+    .childDirectory('runner');
+  if (oldRunnerDirectory.existsSync()) {
+    globals.printStatus(
+      'Delete previous build folder ${oldRunnerDirectory.path}. '
+      'New binaries can be found in ${buildDirectory.childDirectory('runner').path}.'
+    );
+    oldRunnerDirectory.deleteSync(recursive: true);
+  }
```

Comment:
Could we move this logic into `BuildArchitectureMigration`? This would have the benefit of cleaning up the build folder before we run the build, which may help prevent weird issues.

Reply:
Yes, that makes perfect sense!